app/src/main/java/com/example/tasky/feature_authentication/presentation/login_screen/LoginScreen.kt

```diff
@@ -29,56 +31,63 @@ import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.unit.dp
-import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import androidx.navigation.NavController
 import com.example.tasky.R
 import com.example.tasky.feature_authentication.domain.util.AuthUseCaseResult
 import com.example.tasky.feature_authentication.domain.validation.EmailError
 import com.example.tasky.feature_authentication.domain.validation.PasswordError
-import com.example.tasky.feature_authentication.presentation.components.TextFieldComponent
+import com.example.tasky.feature_authentication.presentation.components.TaskyTextField
 import com.example.tasky.presentation.theme.BackgroundBlack
 import com.example.tasky.presentation.theme.BackgroundWhite
 import com.example.tasky.presentation.theme.BtnNavRegScreen
 import com.example.tasky.presentation.theme.LoginBtnTextColor
 import com.example.tasky.presentation.theme.HintColor
+import com.example.tasky.presentation.theme.RedInvalid
 import com.example.tasky.util.Screen
 
 @Composable
 fun LoginScreen(
     navController: NavController,
     viewModel: LoginViewModel = hiltViewModel()
 ) {
-    val state by viewModel.state.collectAsState()
+    val state by viewModel.state.collectAsStateWithLifecycle()
     val context = LocalContext.current
 
-    LaunchedEffect(key1 = viewModel) {
+    var emailError by remember {
+        mutableStateOf("")
+    }
+    var passwordError by remember {
+        mutableStateOf("")
+    }
```

Comment:
These should also be states in the ViewModel, otherwise they won't persist screen rotations. You don't need a channel to send these errors either, since they're persistent and will stay until the user does something (like changing the text)